OpenTrons/opentrons-api
# refactor(robot-server): add PE commands to /sessions responses

## Overview

This PR builds on the models added to the ProtocolEngine by #7993 and adds ProtocolEngine commands to the (currently feature-flagged) `/sessions` endpoints. It also adds two new endpoints:

- `GET /sessions/:session_id/commands`
- `GET /sessions/:session_id/commands/:command_id`

Closes #7871.

## Changelog

- Add `SessionCommandSummary` models to `GET /sessions` and `GET /sessions/:session_id`
- Add new `GET /sessions/:session_id/commands` endpoint that returns `SessionCommandSummary` models
- Add new `GET /sessions/:session_id/commands/:command_id` endpoint that returns full `protocol_engine.Command` models

In real world testing, serializing all the commands of even a simple JSON protocol in full took _a long_ time (1.5s response time on an OT-2). So, after disucssions in Slack, I decided to create a minimal `SessionCommandSummary` model for the collection endpoints. We should expect to revisit this decision as development progresses, but for now I think it's a reasonable compromise.

## Review requests

- [ ] Everything works as advertised (pending test plan)
- [ ] Code looks good
- [ ] Tests look good
- [ ] Generated OpenAPI Spec makes sense

### Smoke test plan

I've updated the [Postman sessions testing collection](https://gist.github.com/mcous/7dde93b6ea83753b86b0d84b40ba07f6) with:

- A simple JSON test protocol (`Simple Test Protocol.json`)
- The new endpoints listed above
- Automatic setting of the `command_id` variable to the first command in the `POST /sessions` response, if present

Using this collection, this is the test smoke test procedure I've been running on the dev server and a real robot:

1. Upload `Simple Test Protocol.json` using a `POST /protocols` request
2. Create a protocol session using a `POST /sessions { sessionType: protocol }` request
    - [ ] Verify command summaries are in `data.commands`
3. Fetch all commands using a `GET /sessions/:session_id/commands` request
    - [ ] Verify command summaries are in `data`
3. Fetch a full command using a `GET /sessions/:session_id/commands/:command_id` request
    - [ ] Verify full command model from ProtocolEngine is in `data`

## Risk assessment

N/A; work is entirely feature flagged
